src/modules/DocGen.py

```diff
@@ -152,7 +167,144 @@ def _read_details(self, member):
 
         return "\n".join(details)
 
+    def _read_proto(self, member):
+        proto = []
+        proto.append(member.find("definition").text)
+
+        proto.append(member.find("argsstring").text)
+        
+        return " ".join(proto)
+
+    def _read_details(self, member):
+        details = []
+        paras = member.findall("para")
+
+        for para in paras:
+            if para.text:
+                details.append(para.text.strip())
+        
+            parameter_list = para.find("parameterlist")
+            if parameter_list is not None and parameter_list.attrib.get("kind") == "param":
+                for param in parameter_list.findall("parameteritem"):
+                    if param.find(".//parametername") is not None:
+                        param_name = param.find(".//parametername").text 
+                    else:
+                        param_name = "Unnamed parameter"
+                    if param.find(".//parameterdescription/para") is not None:
+                        param_desc = param.find(".//parameterdescription/para").text
+                    else:
+                        param_desc = "No description"
+                    details.append(f"Param `{param_name}`: {param_desc}")
+            
+            for simplesect in para.findall("simplesect"):
+                kind = simplesect.attrib.get("kind")
+                sect_text = simplesect.find("para").text if simplesect.find("para") is not None else ""
+                if kind == "return":
+                    details.append(f"Returns: {sect_text}")
+
+        return "\n".join(details)
+
+    def extract_all_paras(self, parent):
+        """
+        Concatenate all <para> elements under the given parent element.
+        
+        This method extracts text from all <para> elements within a parent XML element,
+        concatenating them with newlines. This ensures that multi-paragraph descriptions
+        are captured completely rather than just the first paragraph.
+        
+        Args:
+            parent: XML element that may contain <para> child elements
+            
+        Returns:
+            str: Concatenated text from all <para> elements, or empty string if no parent
+        """
+        if parent is None:
+            return ""
+        return "\n".join(
+            "".join(para.itertext()).strip()
```

Comment:
Should this be

```suggestion
            " ".join(para.itertext()).strip()
```

? (i.e. joining bits of text with a space)